sound/soc/sof/mediatek/mt8195/mt8195-loader.c

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// Copyright (c) 2021 Mediatek Corporation. All rights reserved.
+//
+// Author: YC Hung <yc.hung@mediatek.com>
+//
+// Hardware interface for mt8195 DSP code loader
+
+#include <sound/sof.h>
+#include "mt8195.h"
+#include "../../ops.h"
+
+void hifixdsp_boot_sequence(void *pdev, u32 boot_addr)
+{
+	struct snd_sof_dev *sdev = pdev;
```

Comment:
When you cast it back right away?

Reply:
Do I need to use cast here or I need to modify the parameter as snd_sof_dev *pdev? Thanks

Reply:
I think you should have
```c
void hifixdsp_boot_sequence(struct snd_sof_dev *sdev, u32 boot_addr)
```

Reply:
ok. I will modify it. thanks